kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryption.java

```diff
@@ -100,25 +105,86 @@ private static void checkCipherSuite() {
+                return number.intValue();
+            }
+            else if (value instanceof String stringValue) {
+                return Integer.parseInt(stringValue);
+            }
+            else {
+                throw new IllegalArgumentException("could not convert " + property + " with type " + value.getClass().getSimpleName() + " to Integer");
+            }
+        }).orElse(null);
+    }
+
+    @Nullable
+    private static Long getExperimentalLong(Map<String, Object> experimental, String property) {
+        return Optional.ofNullable(experimental).map(x -> x.get(property)).map(value -> {
+            if (value instanceof Number number) {
+                return number.longValue();
+            }
+            else if (value instanceof String stringValue) {
+                return Long.parseLong(stringValue);
+            }
+            else {
+                throw new IllegalArgumentException("could not convert " + property + " with type " + value.getClass().getSimpleName() + " to Integer");
+            }
+        }).orElse(null);
+    }
+
     @Override
     public SharedEncryptionContext<K, E> initialize(FilterFactoryContext context,
                                                     RecordEncryptionConfig configuration)
             throws PluginConfigurationException {
         checkCipherSuite();
-        KmsService<Object, K, E> kmsPlugin = context.pluginInstance(KmsService.class, configuration.kms());
-        kmsPlugin.initialize(configuration.kmsConfig());
+        var kmsPlugin = context.pluginInstance(KmsService.class, configuration.kms());
+        kmsPlugin.initialize(convertConfigValueToConfigType(kmsPlugin, configuration.kmsConfig()));
         Kms<K, E> kms = buildKms(configuration, kmsPlugin);
 
-        var dekConfig = configuration.dekManager();
+        var dekConfig = dekManager(configuration);
         DekManager<K, E> dekManager = new DekManager<>(kms, dekConfig.maxEncryptionsPerDek());
 
-        KmsCacheConfig cacheConfig = configuration.kmsCache();
+        KmsCacheConfig cacheConfig = kmsCache(configuration);
         EncryptionDekCache<K, E> encryptionDekCache = new EncryptionDekCache<>(dekManager, null, EncryptionDekCache.NO_MAX_CACHE_SIZE,
                 cacheConfig.encryptionDekCacheRefreshAfterWriteDuration(), cacheConfig.encryptionDekCacheExpireAfterWriteDuration());
         DecryptionDekCache<K, E> decryptionDekCache = new DecryptionDekCache<>(dekManager, null, DecryptionDekCache.NO_MAX_CACHE_SIZE);
         return new SharedEncryptionContext<>(kms, kmsPlugin::close, configuration, dekManager, encryptionDekCache, decryptionDekCache);
     }
 
+    private static <P, C> @NonNull C convertConfigValueToConfigType(P plugin, Object conf) {
+        // Get the class of the config type from the plugin instance
+        Plugin annotation = plugin.getClass().getAnnotation(Plugin.class);
+        if (annotation == null) {
+            return (C) conf;
```

Comment:
This return is only really here for a couple of tests, which used mock plugins, whose class lacks the `@Plugin` annotation.

Reply:
I don't think so. `java.lang.Class.getAnnotation` can't be intercepted and I don't think mockito lets you add annotations on to things.

Reply:
This one has gone away now we're generating the annotations.